### PR TITLE
Fixes #28279 - Deprecate Smart Variables

### PR DESCRIPTION
**Pull request by tbrisker**

### CODE REVIEW

File: config/initializers/deprecations.rb [hunk 1]
```diff
@@ -25,3 +25,29 @@
     )
   end
 end
+
+Foreman::Application.configure do |app|
+  config.after_initialize do
+    next if Foreman.in_rake?
+    next unless VariableLookupKey.any?
```
Comment (ekohl) [inline config/initializers/deprecations.rb:59]: Aren't smart class parameters stored in the same database model?

Comment (tbrisker) [reply config/initializers/deprecations.rb:59]: In the same database table, yes, but using STI (single table inheritance). Both models `PuppetClassLookupKey` (smart class params) and `VariableLookupKey` (smart variables) inherit from the `LookupKey` class (as well as ansible variables and possibly others), which is stored on the `lookup_keys` table in the database with `type` column indicating which of the different children classes a row relates to.

**Review by ekohl — APPROVED** [review]

**Comment by ekohl** [comment]
Note I didn't test it, more of an ACK on the concept. The test failures look odd though.

**Comment by tbrisker** [comment]
these look like some rare transient failures i've seen from time to time, but [test foreman] just to be sure

**Comment by tbrisker** [comment]
all is green, no objections raised on discourse, let's merge it.

**Comment by orrabin** [comment]
Thanks @tbrisker !

**Comment by tbrisker** [comment]
1.24 - 7104563ea

**Pull request merged**